core/domain/user_services.py

```diff
@@ -295,6 +298,75 @@ def require_valid_username(cls, username):
                         'This username is not available.')
 
 
+class UserAuth(python_utils.OBJECT):
+    """Value object representing a user's authentication information.
+
+    Attributes:
+        user_id: str. The unique ID of the user.
+        gae_id: str. The ID of the user retrieved from GAE.
+        pin: str or None. The PIN of the user's profile for android.
+        parent_user_id: str or None. User id of the full user to which a
```

Comment:
Change first sentence of description to: `For profile users, the user ID of the full user associated with that profile.`

Ditto elsewhere.

Reply:
Done. Please check if it explains properly why we need to store `parent_user_id` in UserAuthModel from authentication point of view.